app/services/assessors/property_assessor.rb

```diff
@@ -0,0 +1,68 @@
+module Assessors
+  class PropertyAssessor
+    Result = Struct.new(:transaction_allowance,
+                        :allowable_outstanding_mortgage,
+                        :net_value,
+                        :net_equity,
+                        :main_home_equity_disregard,
+                        :assessed_equity,
+                        keyword_init: true)
+    class << self
+      def call(property, remaining_mortgage_allowance, level_of_representation, submission_date)
+        transaction_allowance = calculate_property_transaction_allowance(property, level_of_representation, submission_date)
+        allowable_outstanding_mortgage = calculate_outstanding_mortgage(property, remaining_mortgage_allowance)
+        net_value = calculate_net_value(property, transaction_allowance, allowable_outstanding_mortgage)
+        net_equity = calculate_net_equity(property, net_value)
+        main_home_equity_disregard = calculate_main_home_disregard(property, submission_date)
+        assessed_equity = calculate_assessed_equity(net_equity, main_home_equity_disregard)
+        result = Result.new(transaction_allowance:, allowable_outstanding_mortgage:, net_value:, net_equity:, main_home_equity_disregard:, assessed_equity:)
+        save!(property, result)
+        result
```

Comment:
freezing to make immutable is a useful addition. Shame we can't use Data (Ruby 3.2 addition)